hera_qm/ant_metrics.py

```diff
@@ -0,0 +1,394 @@
+        redCorr = self.red_corr_metrics(pols=['xx','yy'], xants=self.xants, rawMetric=True)
+        meanVijXPol = self.mean_Vij_cross_pol_metrics(xants=self.xants, rawMetric=True)
+        redCorrXPol = self.red_corr_cross_pol_metrics(xants=self.xants, rawMetric=True)   
+        
+        #Save all metrics and zscores
+        metrics, modzScores = {}, {}
+        for metName in ['meanVij','redCorr','meanVijXPol','redCorrXPol']:
+            metric = eval(metName)
+            metrics[metName] = metric
+            modz = per_antenna_modified_z_scores(metric)
+            modzScores[metName] = modz
+            for key in metric.keys():
+                if self.finalMetrics.has_key(metName):
+                    self.finalMetrics[metName][key] = metric[key]
+                    self.finalModzScores[metName][key] = modz[key]
+                else:
+                    self.finalMetrics[metName] = {key: metric[key]}
+                    self.finalModzScores[metName] = {key: modz[key]}
+        self.allMetrics.append(metrics)
+        self.allModzScores.append(modzScores)        
+    
+    def iterative_antenna_metrics_and_flagging(self, crossCut=5, deadCut=5, verbose=False):
+        '''Runs all four metrics (two for dead antennas two for cross-polarized antennas) and saves
+        the results internally to this this antenna metrics object. 
+        
+        Arguments:
+        crossCut -- Modified z-score cut for most cross-polarized antenna. Default 5 "sigmas".
+        deadCut -- Modified z-score cut for most likely dead antenna. Default 5 "sigmas".
+        '''
+        
+        #Summary statistics
+        self.xants, self.crossedAntsRemoved, self.deadAntsRemoved = [], [], []
+        self.removalIter = {}
+        self.allMetrics, self.allModzScores = [], []
+        self.finalMetrics, self.finalModzScores = {}, {}
+        self.crossCut, self.deadCut = crossCut, deadCut
+        
+        #Loop over 
+        for n in range(len(self.antpols)*len(self.ants)):
+            self._run_all_metrics()
+            
+            # Mostly likely dead antenna
+            deadMetrics = average_abs_metrics(self.allModzScores[-1]['meanVij'], 
+                                              self.allModzScores[-1]['redCorr'])
+            worstDeadAnt = max(deadMetrics, key=deadMetrics.get)
+            worstDeadCutRatio = np.abs(deadMetrics[worstDeadAnt])/deadCut
+
+            # Most likely cross-polarized antenna 
+            crossMetrics = average_abs_metrics(self.allModzScores[-1]['meanVijXPol'],
+                                               self.allModzScores[-1]['redCorrXPol'])
+            worstCrossAnt = max(crossMetrics, key=crossMetrics.get)
+            worstCrossCutRatio = np.abs(crossMetrics[worstCrossAnt])/crossCut
+
+            # Find the single worst antenna, remove it, log it, and run again
+            if worstCrossCutRatio >= worstDeadCutRatio and worstCrossCutRatio >= 1.0:
```

Comment:
This doesn't evaluate to true for any of the tests. We should make sure it gets covered, considering it's a significant piece of code.